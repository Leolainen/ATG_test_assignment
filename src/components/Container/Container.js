import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";
import styles from "./styles.module.css";

const Container = (props) => {
  const { className, children, component: Component = "div", ...other } = props;

  return (
    <Component
      className={classnames(styles.root, styles.maxWidth, className)}
      {...other}
    >
      {children}
    </Component>
  );
};

Container.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  component: PropTypes.elementType,
};

Container.uiName = "Container";

export default Container;
