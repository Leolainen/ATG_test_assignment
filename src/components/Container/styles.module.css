.root {
  margin-left: auto;
  margin-right: auto;
  padding-left: 8px;
  padding-right: 8px;

  @media screen and (min-width: 600px) {
    padding-left: 16px;
    padding-right: 16px;
  }
}

.maxWidth {
  max-width: 788px;
}
