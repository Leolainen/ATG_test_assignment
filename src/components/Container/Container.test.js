import React from "react";
import { render, cleanup } from "@testing-library/react";
import Container from "./Container";

describe("<Container />", () => {
  afterEach(cleanup);

  describe("Snapshots", () => {
    it("should render", () => {
      const { asFragment } = render(<Container>container</Container>);

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
