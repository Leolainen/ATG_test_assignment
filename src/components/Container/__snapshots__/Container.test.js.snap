// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> Snapshots should render 1`] = `
<DocumentFragment>
  <div
    class="root maxWidth"
  >
    container
  </div>
</DocumentFragment>
`;
